subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/InputPropertySpec.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.properties;
+
+import org.gradle.api.internal.tasks.ValidatingValue;
+
+public class InputPropertySpec implements PropertySpec {
+    private final String propertyName;
+    private final ValidatingValue value;
```

Comment:
I put `propertyName` in an abstract superclass. I didn't put `value` there, since the file specs don't have a `ValidatingValue` but only a `FileCollection`.